lang/java/avro/src/main/java/org/apache/avro/file/SnappyCodec.java

```diff
@@ -37,7 +40,7 @@ static class Option extends CodecFactory {
 
     @Override
     protected Codec createInstance() {
-      return new SnappyCodec();
+      return SnappyCodec.INSTANCE;
```

Comment:
Hello,
A comment and a question :
As SnappyCodec contains a crc32 field that change [its state on Compress call](https://github.com/apache/avro/blob/master/lang/java/avro/src/main/java/org/apache/avro/file/SnappyCodec.java#L57-L58), usage of a singleton will lead to error in multi thread environment.
On your issue, do you have any log that contains "Snappy was not available". That could explain why "snappy" is null [with this code](https://github.com/apache/avro/blob/master/lang/java/avro/src/main/java/org/apache/avro/file/CodecFactory.java#L68-L72) ?

Reply:
For related descriptions, see:
https://issues.apache.org/jira/browse/AVRO-3590